### DISCUSSION TITLE
Issues with idom.run and SharedClientStateServer

### DISCUSSION THREAD

**Discussion by rmorshea** [idom-team/idom#711] [General]
@jmtaysom was having some issues with `idom.run` when using Sanic as well as with Sanic's `SharedClientStateServer` (previously discussed [here](https://github.com/idom-team/idom/pull/710#issuecomment-1069788681)).

**Comment by rmorshea** [comment]
@jmtaysom what version of Sanic are you using?

**Reply by jmtaysom** [reply]
Sanic 21.12.1

**Reply by rmorshea** [reply]
IDOM presently requires [`sanic>19,<19.12`](https://github.com/idom-team/idom/blob/main/requirements/pkg-extras.txt#L2) - see [here](https://github.com/idom-team/idom/issues/678) for more info. The other async server implementations should work fine without this limited range of versions.

Again, apologies for all these limitations. I'm presently working on [a PR](https://github.com/idom-team/idom/pull/703) which will address most of the problems we've talked about:

- Rework `idom.run`
- Remove `SharedClientStateServer`
- Fix limited support for `sanic`
- etc.

**Comment by rmorshea** [comment]
Also, apologies for the sweeping changes here, but `SharedClientStateServer` is slated for removal since it doesn't work with multi-process web servers. Instead we would need to provide a more robust solution that would work across processes and hosts - probably using something like Redis.

**Reply by jmtaysom** [reply]
That makes a lot of sense. If it is slated for removal I would remove it from the beginning of the tutorial just to not confuse people.
